service/awsconfig/v10/resource/ebsvolume/current.go

```diff
@@ -23,53 +19,15 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	currentState, err := r.persistentVolumes(customObject)
+	// Get both the Etcd volume and any Persistent Volumes.
+	volumes, err := r.service.ListVolumes(customObject, true, true)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	return currentState, nil
-}
-
-func (r *Resource) persistentVolumes(customObject v1alpha1.AWSConfig) (*EBSVolumeState, error) {
-	volumeState := &EBSVolumeState{}
-	volumeIDs := []string{}
-
-	// We filter to only select clusters with the cluster cloud provider tag.
-	clusterTag := key.ClusterCloudProviderTag(customObject)
-	filters := []*ec2.Filter{
-		{
-			Name: aws.String(fmt.Sprintf("tag:%s", clusterTag)),
-			Values: []*string{
-				aws.String(cloudProviderClusterTagValue),
-			},
-		},
-	}
-	output, err := r.clients.EC2.DescribeVolumes(&ec2.DescribeVolumesInput{
-		Filters: filters,
-	})
-	if err != nil {
-		return nil, microerror.Mask(err)
+	currentState := &EBSVolumeState{
+		Volumes: volumes,
 	}
 
-	for _, vol := range output.Volumes {
-		// Volume is only included if it has a PV name tag.
-		if containsPersistentVolumeTag(vol.Tags) {
-			volumeIDs = append(volumeIDs, *vol.VolumeId)
-		}
-	}
-
-	volumeState.VolumeIDs = volumeIDs
-
-	return volumeState, nil
-}
-
-func containsPersistentVolumeTag(tags []*ec2.Tag) bool {
-	for _, tag := range tags {
-		if *tag.Key == cloudProviderPersistentVolumeTagKey {
-			return true
-		}
-	}
-
-	return false
+	return currentState, nil
```

Comment:
The diff is not useful but the implementation looks nicer now. Above are some constants that are not used in that file. Maybe it is dead code or belongs somewhere else.